tabular/src/autogluon/tabular/testing/fit_helper.py

```diff
@@ -0,0 +1,456 @@
+        return DatasetLoaderHelper.toy_map[name]()
+
+    @staticmethod
+    def load_dataset(name: str, directory_prefix: str = "./datasets/") -> tuple[pd.DataFrame, pd.DataFrame, dict]:
+        if name in DatasetLoaderHelper.toy_map:
+            return DatasetLoaderHelper.load_dataset_toy(name=name)
+        dataset_info = copy.deepcopy(DatasetLoaderHelper.dataset_info_dict[name])
+        train_file = dataset_info.pop("train_file", "train_data.csv")
+        test_file = dataset_info.pop("test_file", "test_data.csv")
+        name_inner = dataset_info.pop("name")
+        url = dataset_info.pop("url", None)
+        train_data, test_data = DatasetLoaderHelper.load_data(
+            directory_prefix=directory_prefix,
+            train_file=train_file,
+            test_file=test_file,
+            name=name_inner,
+            url=url,
+        )
+
+        return train_data, test_data, dataset_info
+
+    @staticmethod
+    def load_data(
+        directory_prefix: str,
+        train_file: str,
+        test_file: str,
+        name: str,
+        url: str | None = None,
+    ):
+        if not os.path.exists(directory_prefix):
+            os.mkdir(directory_prefix)
+        directory = directory_prefix + name + "/"
+        train_file_path = directory + train_file
+        test_file_path = directory + test_file
+        if (not os.path.exists(train_file_path)) or (not os.path.exists(test_file_path)):
+            # fetch files from s3:
+            print("%s data not found locally, so fetching from %s" % (name, url))
+            zip_name = download(url, directory_prefix)
+            unzip(zip_name, directory_prefix)
+            os.remove(zip_name)
+
+        train_data = TabularDataset(train_file_path)
+        test_data = TabularDataset(test_file_path)
+        return train_data, test_data
+
+
+class FitHelper:
+    @staticmethod
+    def fit_and_validate_dataset(
+        dataset_name: str,
+        fit_args: dict[str, Any],
+        init_args: dict[str, Any] | None = None,
+        sample_size: int | None = 1000,  # FIXME: default to None
+        refit_full: bool = True,
+        delete_directory: bool = True,
+        extra_metrics: list[str] | None = None,
+        extra_info: bool = False,
+        predictor_info: bool = False,
+        expected_model_count: int | None = 2,
+        fit_weighted_ensemble: bool = True,
+        min_cls_count_train: int = 1,
+        path_as_absolute: bool = False,
+        compile: bool = False,
+        compiler_configs: dict | None = None,
+        allowed_dataset_features: list[str] | None = None,
+        expected_stacked_overfitting_at_test: bool | None = None,
+        expected_stacked_overfitting_at_val: bool | None = None,
+        scikit_api: bool = False,
+        use_test_data: bool = False,
+        use_test_for_val: bool = False,
+        raise_on_model_failure: bool | None = None,
+        deepcopy_fit_args: bool = True,
```

Comment:
Is there a reason not to always do the deep copy?

Reply:
Yes: For callback objects, we want to check the state of the callback after fitting, so avoiding deepcopy lets us view the callback object after updates to ensure the logic is working correctly.